hash-fetch/src/urlhint.rs

```diff
@@ -173,36 +171,30 @@ fn decode_urlhint_output(output: (String, H160, Address)) -> Option<URLHintResul
 }
 
 impl URLHint for URLHintContract {
-	fn resolve(&self, id: Bytes) -> BoxFuture<Option<URLHintResult>, String> {
-		use futures::future::Either;
-
-		let do_call = |_, data| {
-			let addr = match self.client.registrar() {
-				Ok(addr) => addr,
-				Err(e) => return Box::new(future::err(e))
-					as BoxFuture<Vec<u8>, _>,
-			};
-
-			self.client.call(addr, data)
+	fn resolve(&self, id: H256) -> Box<Future<Item = Option<URLHintResult>, Error = String> + Send> {
+		let address = match self.client.registrar() {
+			Ok(a) => a,
+			Err(e) => return Box::new(future::err(e)),
 		};
 
-		let urlhint = self.urlhint.clone();
 		let client = self.client.clone();
-		Box::new(self.registrar.get_address(do_call, keccak("githubhint"), "A".into())
-			.map(|addr| if addr == Address::default() { None } else { Some(addr) })
-			.and_then(move |address| {
-				let mut fixed_id = [0; 32];
-				let len = ::std::cmp::min(32, id.len());
-				fixed_id[..len].copy_from_slice(&id[..len]);
-
-				match address {
-					None => Either::A(future::ok(None)),
-					Some(address) => {
-						let do_call = move |_, data| client.call(address, data);
-						Either::B(urlhint.entries(do_call, H256(fixed_id)).map(decode_urlhint_output))
-					}
-				}
-			}))
+		let get_address = self.registrar.functions().get_address();
+		let entries = self.urlhint.functions().entries();
+		let data = get_address.input(keccak("githubhint"), "A");
+
+		let future = client.call(address, data)
```

Comment:
Why is it not using `get_address.call`?

Reply:
cause `client.call` returns a future and async calls and not supported in the current version of `ethabi_derive`. afaik, that's the only place where async calls where actually used in parity